mops/CMakeLists.txt

```diff
@@ -107,7 +107,13 @@ if(CMAKE_CUDA_COMPILER)
     target_compile_definitions(mops PUBLIC MOPS_CUDA_ENABLED)
     target_sources(mops
         PRIVATE
-            src/opsa/opsa.cu
+        "src/utils/cuda_utils.cu"
+        "include/mops/cuda_utils.cuh"
```

Comment:
since this is a private header, I'd put it in `src/utils/cuda_utils.cuh`

Reply:
any CUDA code is going to require `cuda_utils.cuh`, since that contains the shared memory helper functions.

Reply:
sure, but this is not part of the API of mops, right? All the CUDA implementations can do `#include "utils/cuda_utils.cuh"`, but external users of the library should not have access to these functions unless I'm missing something.